block-modes/src/errors.rs

```diff
@@ -11,27 +11,34 @@ pub struct BlockModeError;
 pub struct InvalidKeyIvLength;
 
 impl fmt::Display for BlockModeError {
-    fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
-        f.write_str("BlockModeError")
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "BlockModeError")
```

Comment:
Since we don't have any formatting arguments, I think using `f.write_str()` is perfectly fine. Otherwise with `write!` macro we rely on compiler to optimize out all unnecessary layers.